File: Android/sources/com/google/android/material/timepicker/g.java
package com.google.android.material.timepicker;

import android.content.res.Resources;
import android.text.InputFilter;
import android.text.TextUtils;
import android.view.View;
import android.view.inputmethod.InputMethodManager;
import android.widget.EditText;
import android.widget.LinearLayout;
import android.widget.TextView;
import com.google.android.material.button.MaterialButtonToggleGroup;
import com.google.android.material.textfield.TextInputLayout;
import java.util.Arrays;
import java.util.Locale;

public final class g implements qjf, mjf {
    public final LinearLayout a;
    public final ijf b;
    public final ojf c;
    public final ojf o;
    public final ChipTextInputComboView v;
    public final ChipTextInputComboView w;
    public final EditText x;
    public final EditText y;
    public final MaterialButtonToggleGroup z;

    public g(LinearLayout linearLayout, ijf ijf) {
        ojf ojf = new ojf(0, this);
        this.c = ojf;
        ojf ojf2 = new ojf(1, this);
        this.o = ojf2;
        this.a = linearLayout;
        this.b = ijf;
        Resources resources = linearLayout.getResources();
        ChipTextInputComboView chipTextInputComboView = (ChipTextInputComboView) linearLayout.findViewById(agc.material_minute_text_input);
        this.v = chipTextInputComboView;
        ChipTextInputComboView chipTextInputComboView2 = (ChipTextInputComboView) linearLayout.findViewById(agc.material_hour_text_input);
        this.w = chipTextInputComboView2;
        ((TextView) chipTextInputComboView.findViewById(agc.material_label)).setText(resources.getString(ymc.material_timepicker_minute));
        ((TextView) chipTextInputComboView2.findViewById(agc.material_label)).setText(resources.getString(ymc.material_timepicker_hour));
        chipTextInputComboView.setTag(agc.selection_type, 12);
        chipTextInputComboView2.setTag(agc.selection_type, 10);
        if (ijf.c == 0) {
            MaterialButtonToggleGroup materialButtonToggleGroup = (MaterialButtonToggleGroup) linearLayout.findViewById(agc.material_clock_period_toggle);
            this.z = materialButtonToggleGroup;
            materialButtonToggleGroup.c.add(new njf(this));
            this.z.setVisibility(0);
            f();
        }
        wld wld = new wld(14, this);
        chipTextInputComboView2.setOnClickListener(wld);
        chipTextInputComboView.setOnClickListener(wld);
        EditText editText = chipTextInputComboView2.c;
        InputFilter[] filters = editText.getFilters();
        InputFilter[] inputFilterArr = (InputFilter[]) Arrays.copyOf(filters, filters.length + 1);
        inputFilterArr[filters.length] = ijf.b;
        editText.setFilters(inputFilterArr);
        EditText editText2 = chipTextInputComboView.c;
        InputFilter[] filters2 = editText2.getFilters();
        InputFilter[] inputFilterArr2 = (InputFilter[]) Arrays.copyOf(filters2, filters2.length + 1);
        inputFilterArr2[filters2.length] = ijf.a;
        editText2.setFilters(inputFilterArr2);
        TextInputLayout textInputLayout = chipTextInputComboView2.b;
        EditText editText3 = textInputLayout.getEditText();
        this.x = editText3;
        TextInputLayout textInputLayout2 = chipTextInputComboView.b;
        EditText editText4 = textInputLayout2.getEditText();
        this.y = editText4;
        f fVar = new f(chipTextInputComboView2, chipTextInputComboView, ijf);
        gag.j(chipTextInputComboView2.a, new pjf(linearLayout.getContext(), ymc.material_hour_selection, ijf, 0));
        gag.j(chipTextInputComboView.a, new pjf(linearLayout.getContext(), ymc.material_minute_selection, ijf, 1));
        editText3.addTextChangedListener(ojf2);
        editText4.addTextChangedListener(ojf);
        e(ijf);
        EditText editText5 = textInputLayout.getEditText();
        EditText editText6 = textInputLayout2.getEditText();
        editText5.setImeOptions(268435461);
        editText6.setImeOptions(268435462);
        editText5.setOnEditorActionListener(fVar);
        editText5.setOnKeyListener(fVar);
        editText6.setOnKeyListener(fVar);
    }

    public final void a() {
        InputMethodManager inputMethodManager;
        LinearLayout linearLayout = this.a;
        View focusedChild = linearLayout.getFocusedChild();
        if (!(focusedChild == null || (inputMethodManager = (InputMethodManager) fw3.b(focusedChild.getContext(), InputMethodManager.class)) == null)) {
            inputMethodManager.hideSoftInputFromWindow(focusedChild.getWindowToken(), 0);
        }
        linearLayout.setVisibility(8);
    }

    public final void b() {
        this.a.setVisibility(0);
        c(this.b.w);
    }

    public final void c(int i) {
        this.b.w = i;
        boolean z2 = false;
        this.v.setChecked(i == 12);
        if (i == 10) {
            z2 = true;
        }
        this.w.setChecked(z2);
        f();
    }

    public final void d() {
        ijf ijf = this.b;
        boolean z2 = false;
        this.v.setChecked(ijf.w == 12);
        if (ijf.w == 10) {
            z2 = true;
        }
        this.w.setChecked(z2);
    }

    public final void e(ijf ijf) {
        EditText editText = this.x;
        ojf ojf = this.o;
        editText.removeTextChangedListener(ojf);
        EditText editText2 = this.y;
        ojf ojf2 = this.c;
        editText2.removeTextChangedListener(ojf2);
        Locale locale = this.a.getResources().getConfiguration().locale;
        String format = String.format(locale, "%02d", new Object[]{Integer.valueOf(ijf.v)});
        String format2 = String.format(locale, "%02d", new Object[]{Integer.valueOf(ijf.b())});
        ChipTextInputComboView chipTextInputComboView = this.v;
        String a2 = ijf.a(chipTextInputComboView.getResources(), format, "%02d");
        chipTextInputComboView.a.setText(a2);
        if (!TextUtils.isEmpty(a2)) {
            a aVar = chipTextInputComboView.o;
            EditText editText3 = chipTextInputComboView.c;
            editText3.removeTextChangedListener(aVar);
            editText3.setText(a2);
            editText3.addTextChangedListener(aVar);
        }
        ChipTextInputComboView chipTextInputComboView2 = this.w;
        String a3 = ijf.a(chipTextInputComboView2.getResources(), format2, "%02d");
        chipTextInputComboView2.a.setText(a3);
        if (!TextUtils.isEmpty(a3)) {
            a aVar2 = chipTextInputComboView2.o;
            EditText editText4 = chipTextInputComboView2.c;
            editText4.removeTextChangedListener(aVar2);
            editText4.setText(a3);
            editText4.addTextChangedListener(aVar2);
        }
        editText.addTextChangedListener(ojf);
        editText2.addTextChangedListener(ojf2);
        f();
    }

    public final void f() {
        MaterialButtonToggleGroup materialButtonToggleGroup = this.z;
        if (materialButtonToggleGroup != null) {
            materialButtonToggleGroup.b(this.b.x == 0 ? agc.material_clock_period_am_button : agc.material_clock_period_pm_button, true);
        }
    }

    public final void invalidate() {
        e(this.b);
    }
}

File: IOS/v2/com/google/android/material/timepicker/g.swift
package com.google.iOS.material.timepicker;.swift
import iOS.content.res.Resources;.swift
import iOS.text.InputFilter;.swift
import iOS.text.TextUtils;.swift
import iOS.view.View;.swift
import iOS.view.inputmethod.InputMethodManager;.swift
import iOS.widget.EditText;.swift
import iOS.widget.LinearLayout;.swift
import iOS.widget.TextView;.swift
import com.google.iOS.material.button.MaterialButtonToggleGroup;.swift
import com.google.iOS.material.textfield.TextInputLayout;.swift
import java.util.Arrays;.swift
import java.util.Locale;.swift
public final class g implements qjf, mjf {.swift
public final LinearLayout a;.swift
public final ijf b;.swift
public final ojf c;.swift
public final ojf o;.swift
public final ChipTextInputComboView v;.swift
public final ChipTextInputComboView w;.swift
public final EditText x;.swift
public final EditText y;.swift
public final MaterialButtonToggleGroup z;.swift
public g(LinearLayout linearLayout, ijf ijf) {.swift
ojf ojf = new ojf(0, this);.swift
this.c = ojf;.swift
ojf ojf2 = new ojf(1, this);.swift
this.o = ojf2;.swift
this.a = linearLayout;.swift
this.b = ijf;.swift
Resources resources = linearLayout.getResources();.swift
ChipTextInputComboView chipTextInputComboView = (ChipTextInputComboView) linearLayout.findViewById(agc.material_minute_text_input);.swift
this.v = chipTextInputComboView;.swift
ChipTextInputComboView chipTextInputComboView2 = (ChipTextInputComboView) linearLayout.findViewById(agc.material_hour_text_input);.swift
this.w = chipTextInputComboView2;.swift
((TextView) chipTextInputComboView.findViewById(agc.material_label)).setText(resources.getString(ymc.material_timepicker_minute));.swift
((TextView) chipTextInputComboView2.findViewById(agc.material_label)).setText(resources.getString(ymc.material_timepicker_hour));.swift
chipTextInputComboView.setTag(agc.selection_type, 12);.swift
chipTextInputComboView2.setTag(agc.selection_type, 10);.swift
if (ijf.c == 0) {.swift
MaterialButtonToggleGroup materialButtonToggleGroup = (MaterialButtonToggleGroup) linearLayout.findViewById(agc.material_clock_period_toggle);.swift
this.z = materialButtonToggleGroup;.swift
materialButtonToggleGroup.c.add(new njf(this));.swift
this.z.setVisibility(0);.swift
f();.swift
}.swift
wld wld = new wld(14, this);.swift
chipTextInputComboView2.setOnClickListener(wld);.swift
chipTextInputComboView.setOnClickListener(wld);.swift
EditText editText = chipTextInputComboView2.c;.swift
InputFilter[] filters = editText.getFilters();.swift
InputFilter[] inputFilterArr = (InputFilter[]) Arrays.copyOf(filters, filters.length + 1);.swift
inputFilterArr[filters.length] = ijf.b;.swift
editText.setFilters(inputFilterArr);.swift
EditText editText2 = chipTextInputComboView.c;.swift
InputFilter[] filters2 = editText2.getFilters();.swift
InputFilter[] inputFilterArr2 = (InputFilter[]) Arrays.copyOf(filters2, filters2.length + 1);.swift
inputFilterArr2[filters2.length] = ijf.a;.swift
editText2.setFilters(inputFilterArr2);.swift
TextInputLayout textInputLayout = chipTextInputComboView2.b;.swift
EditText editText3 = textInputLayout.getEditText();.swift
this.x = editText3;.swift
TextInputLayout textInputLayout2 = chipTextInputComboView.b;.swift
EditText editText4 = textInputLayout2.getEditText();.swift
this.y = editText4;.swift
f fVar = new f(chipTextInputComboView2, chipTextInputComboView, ijf);.swift
gag.j(chipTextInputComboView2.a, new pjf(linearLayout.getContext(), ymc.material_hour_selection, ijf, 0));.swift
gag.j(chipTextInputComboView.a, new pjf(linearLayout.getContext(), ymc.material_minute_selection, ijf, 1));.swift
editText3.addTextChangedListener(ojf2);.swift
editText4.addTextChangedListener(ojf);.swift
e(ijf);.swift
EditText editText5 = textInputLayout.getEditText();.swift
EditText editText6 = textInputLayout2.getEditText();.swift
editText5.setImeOptions(268435461);.swift
editText6.setImeOptions(268435462);.swift
editText5.setOnEditorActionListener(fVar);.swift
editText5.setOnKeyListener(fVar);.swift
editText6.setOnKeyListener(fVar);.swift
}.swift
public final void a() {.swift
InputMethodManager inputMethodManager;.swift
LinearLayout linearLayout = this.a;.swift
View focusedChild = linearLayout.getFocusedChild();.swift
if ((focusedChild == null || (inputMethodManager = (InputMethodManager) fw3.b(focusedChild.getContext(), InputMethodManager.class)) == null)) {.swift
inputMethodManager.hideSoftInputFromWindow(focusedChild.getWindowToken(), 0);.swift
}.swift
linearLayout.setVisibility(8);.swift
}.swift
public final void b() {.swift
this.a.setVisibility(0);.swift
c(this.b.w);.swift
}.swift
public final void c(int i) {.swift
this.b.w = i;.swift
boolean z2 = false;.swift
this.v.setChecked(i == 12);.swift
if (i == 10) {.swift
z2 = true;.swift
}.swift
this.w.setChecked(z2);.swift
f();.swift
}.swift
public final void d() {.swift
ijf ijf = this.b;.swift
boolean z2 = false;.swift
this.v.setChecked(ijf.w == 12);.swift
if (ijf.w == 10) {.swift
z2 = true;.swift
}.swift
this.w.setChecked(z2);.swift
}.swift
public final void e(ijf ijf) {.swift
EditText editText = this.x;.swift
ojf ojf = this.o;.swift
editText.removeTextChangedListener(ojf);.swift
EditText editText2 = this.y;.swift
ojf ojf2 = this.c;.swift
editText2.removeTextChangedListener(ojf2);.swift
Locale locale = this.a.getResources().getConfiguration().locale;.swift
String format = String.format(locale, "%02d", new Object[]{Integer.valueOf(ijf.v)});.swift
String format2 = String.format(locale, "%02d", new Object[]{Integer.valueOf(ijf.b())});.swift
ChipTextInputComboView chipTextInputComboView = this.v;.swift
String a2 = ijf.a(chipTextInputComboView.getResources(), format, "%02d");.swift
chipTextInputComboView.a.setText(a2);.swift
if (TextUtils.isEmpty(a2)) {.swift
a aVar = chipTextInputComboView.o;.swift
EditText editText3 = chipTextInputComboView.c;.swift
editText3.removeTextChangedListener(aVar);.swift
editText3.setText(a2);.swift
editText3.addTextChangedListener(aVar);.swift
}.swift
ChipTextInputComboView chipTextInputComboView2 = this.w;.swift
String a3 = ijf.a(chipTextInputComboView2.getResources(), format2, "%02d");.swift
chipTextInputComboView2.a.setText(a3);.swift
if (TextUtils.isEmpty(a3)) {.swift
a aVar2 = chipTextInputComboView2.o;.swift
EditText editText4 = chipTextInputComboView2.c;.swift
editText4.removeTextChangedListener(aVar2);.swift
editText4.setText(a3);.swift
editText4.addTextChangedListener(aVar2);.swift
}.swift
editText.addTextChangedListener(ojf);.swift
editText2.addTextChangedListener(ojf2);.swift
f();.swift
}.swift
public final void f() {.swift
MaterialButtonToggleGroup materialButtonToggleGroup = this.z;.swift
if (materialButtonToggleGroup = null) {.swift
materialButtonToggleGroup.b(this.b.x == 0 ? agc.material_clock_period_am_button : agc.material_clock_period_pm_button, true);.swift
}.swift
}.swift
public final void invalidate() {.swift
e(this.b);.swift
}.swift
}.swift
